addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1099,21 +1105,23 @@ var SnippetEditor = Widget.extend({
         // Remove the siblings that belong to a snippet in grid mode
         // and put the identified grid mode snippets in their own "selector".
         const selectorGrids = new Set();
-        if (rowEl.classList.contains('row')) {
-            if ($selectorSiblings) {
-                // Looping backwards because elements are removed, so the
-                // indexes are not lost.
-                for (let i = $selectorSiblings.length - 1; i >= 0; i--) {
-                    if (isPopup && !$selectorSiblings[i].closest('div.s_popup')) {
-                        // Removing the siblings that are outside the popup if
-                        // the grid item is in a popup.
+        if (rowEl.classList.contains('row') && $selectorSiblings) {
+            // The grid from which the dragged column comes (if any).
+            const targetGridEl = rowEl.classList.contains('o_grid_mode') ? rowEl : rowEl.closest('.o_grid_mode');
+            // Looping backwards because elements are removed, so the
+            // indexes are not lost.
+            for (let i = $selectorSiblings.length - 1; i >= 0; i--) {
+                if (isPopup && !$selectorSiblings[i].closest('div.s_popup')) {
+                    // Removing the siblings that are outside the popup if
+                    // the grid item is in a popup.
+                    $selectorSiblings.splice(i, 1);
```

Comment:
Relooking at the whole code, I am actually confused: why is there specific code related to removing dropzones in popups elements, only if "`rowEl` is a row" (btw strange to have a variable named `rowEl` which might not be a row). Are not all "selector siblings" element outside of the popup removed if a popup is opened? And should not that code be in website? (I know lots of grid mode stuff is in web_editor instead of website because it would have been a nightmare to split + it could make sense in web_editor at some point but this s_popup thing seems unrelated).

(This comment should not impact this PR, could be something to improve in another PR ;))

Reply:
For the `rowEl` variable name, I know it bugs me too. It was to have the parallel with `columnEl`. But is true that I could have just named it `parentEl` and declared a `rowEl` only when necessary.

For the popup: when I added the code to filter the siblings (without the popup case), I placed it there because it was before the call to `activateInsertionZones` because I wanted to avoid this function to make useless operations on siblings that would be removed anyway. And when I tested with the popup afterwards, it was not working well so I just added this case (it was before you told me to not put things related to website in web_editor so I didn't know I shouldn't). But now that I look again, I should have placed it at the start of `activateInsertionZones`, after the part managing the modals.

I will make an other PR to fix these two issues :)